pnp4nagios update for php5.6 to php8.0

celane opened the pull request:
Major changes: update for php8.0
packaging (make rpms easier ... see spec file in dist and README)
file lock in process_perfdata.pl to avoid some errors.

rrdtools needs env setting XDG_CACHE_HOME for fontconfig
Tested on Fedora 35 (php 8.0) Alma 8.6 (php 7.2) and Fedora 22 (php 5.6).

Commit:
utf8.php and data.php changes for php8
other files to make rpm building work better

Commit:
add systemd startup for npcd spec file edits

Commit:
rrdtool.php now sets env XDG_CACHE_HOME for rrdtool, to
avoid fontconfig crashes.

Commit:
more php8 fixes, systemd file for npcd
note that pid files moved from /var/run  to /run, in line
with modern linux

Commit:
fix to 'basket' display

Commit:
pdf generation fix

Commit:
increment release number

Commit:
magic-quotes was deprecated (php 5?) and removed php7.4? so take out

Commit:
mod to process_perfdata.pl to reduce annoying temp errors
remove rrd_convert and rrd_modify, because they do not work
for current rrd xml configuration.

Commit:
remove various compile warnings from modern gcc

Commit:
typo fix

Commit:
add file locking for process_perfdata stats file updates

Commit:
fix perl

Commit:
Fix bugs with process_perfdata
file locking is needed for the few times/day when
a pair of process_perfdata runs try to update the stats
file at the same time; minor but annoying.

Updated release version in .spec file

Commit:
update spec file, added a dependency

Commit:
change build configuration to use git-build-rpm package

Commit:
fixes

Commit:
update to README regarding rpm build instructions

Commit:
more minor packaging stuff

Commit:
more packaging improvements

Commit:
fix internal stats race in perfdata.pl
add logwatch config setup
spec file update

Commit:
logwatch config files

Commit:
fix prot on logwatch executable

Commit:
minor fix to logwatch script

Commit:
fix logwatch script permissions *again*

russellvt reviewed (changes requested):
This should be broken apart intop smaller changes (Read: It's not just "PHP5-PHP8" migration).

Some ideas for general categories:
* Fix RPM Build from `git`
* Various Fedora contribs:
** Logwatch for `pnp4nagios`
** `npcd` system/service/startup configurations and scripts
** Logrotate configuration
* Fixes `rpmbuild` from `git`
* File locking for process_perfdata
** Is this actually needed? Is there a race/contention there, right now, between threads?

Note: This also appears to break / change some things / nehaviour in potentially unexpected ways on other platforms. Some examples:

* Should likely remove Subversion tags sich as `$Id$` (in-comments, essentially harmless)
* Removes previously used environment variables and adds others
* Hard-coded non-standard (?) directories (eg. /run instead of /var/run)

Overall, I appreciate the contirbution!

It would be nice to break this apart into smaller bite-size quantites, though (all of the Fedora contrib, for example). Most of this isn't a "PHP8 migration," unfortunately.

celane committed:
fixes for dealing with "deprecated" features throwing errors in php 8.2

celane committed:
php 8.2 deprecates turning a "false" variable into an array,
so have to do it explicitly.
Also, set up environment variable XDG_CACHE_HOME in systemd startup
of httpd with a plugin

celane committed:
update Changelog

celane committed:
Merge branch 'russellvt:master' into master

celane committed:
typical php8.2 fixes for docs.php

celane committed:
Merge branch 'master' of https://github.com/celane/pnp4nagios

celane committed:
increment release number

celane committed:
Terminate pid file with newline; startd complaining.

celane committed:
small updates to configure (warnings, version #)

celane committed:
Update ajax.php

php8.1+ usual complaints about auto-creating arrays

celane committed:
Update README.md
